nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3BinaryManager.java

```diff
@@ -311,10 +307,11 @@ protected void setupCloudClient() throws IOException {
             try {
                 // Open keystore
                 File ksFile = new File(keystoreFile);
-                FileInputStream ksStream = new FileInputStream(ksFile);
-                KeyStore keystore = KeyStore.getInstance(KeyStore.getDefaultType());
-                keystore.load(ksStream, keystorePass.toCharArray());
-                ksStream.close();
+                KeyStore keystore;
+                try (FileInputStream ksStream = new FileInputStream(ksFile)) {
+                    keystore = KeyStore.getInstance(KeyStore.getDefaultType());
+                    keystore.load(ksStream, keystorePass.toCharArray());
+                }
                 // Get keypair for alias
                 if (!keystore.isKeyEntry(privkeyAlias)) {
```

Comment:
risk of NPE?

Reply:
Nowpe, because KeyStore.getInstance() does return null but throws an exception